### PR TITLE
Pass options from User.login to createAccessToken

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 

It will allow subclass of User to create access token based on additional properties such as 'scope'.

### CODE REVIEW

File: common/models/user.js [hunk 1]
```diff
@@ -69,11 +69,23 @@ module.exports = function(User) {
    * customize how access tokens are generated
    *
    * @param {Number} ttl The requested ttl
+   * @param {Object} options The options for access token
```
Comment (ritch) [inline common/models/user.js]: This could be a bit clearer.

Comment (ritch) [inline common/models/user.js]: You should also update https://github.com/strongloop/loopback/pull/1152/files#diff-5d81af95449574544e7317f3eda2badeL144 with some docs for this.

Comment (raymondfeng) [inline common/models/user.js]: I assume you mean to update docs of User.login at docs.strongloop.com.

File: common/models/user.js [hunk 2]
```diff
@@ -210,19 +236,11 @@ module.exports = function(User) {
...
-                  //  1. token.user already exists, it's a function injected by
-                  //     "AccessToken belongsTo User" relation
-                  //  2. ModelBaseClass.toJSON() ignores own properties, thus
-                  //     the value won't be included in the HTTP response
-                  // See also loopback#161 and loopback#162
-                  token.__data.user = user;
-                }
-                fn(err, token);
-              });
+              if (user.createAccessToken.length === 2) {
```
Comment (seriousben) [inline common/models/user.js]: @raymondfeng Not sure what this means? How can I pass credentials to to createAccessToken?

Comment (seriousben) [inline common/models/user.js]: NVM, just learned something new.

**Pull request merged**